Swift/Collection.swift

```diff
@@ -302,6 +303,42 @@ public final class Collection : CollectionChangeObservable, Indexable, Equatable
         return ListenerToken(token)
     }
     
+    // MARK: Combine Publisher
+    
+    @available(iOS 13.0, *)
+    public func changePublisher(on queue: DispatchQueue = .main) -> AnyPublisher<CollectionChange, Never> {
+        let subject = PassthroughSubject<CollectionChange, Never>()
+        var token: ListenerToken?
+        
+        token = self.addChangeListener(queue: queue) { change in
+            subject.send(change)
+        }
+
+        return subject
+            .receive(on: queue)
+            .handleEvents(receiveCompletion: { _ in
```

Comment:
I'm not 100% on this but from the [doc](https://developer.apple.com/documentation/combine/passthroughsubject/handleevents(receivesubscription:receiveoutput:receivecompletion:receivecancel:receiverequest:), should it handles `receivedCancel` event to remove the token instead?

Reply:
Probably should have the test case to test this.

Reply:
Both should be there if we want to handle both cases. I haven't even considered this... how would a listener for collection/document would be "canceled"? I couldn't think of an example that would refer to the existing implementation as canceled. But as we do allow for the token to be removed AT ANY TIME... I guess we need to handle all "such" scenarios - which includes a "cancel" event on the publisher.

Reply:
Only receiveCancel is needed here because there is no upstream publisher for the PassthroughSubject created here to send the complete event. The receiveCancel event is what the downstream subscribers send whey they call cancel() on the returned Cancellable. 

As discussed in the meeting yesterday, the PassthroughSubject can send the complete event to the downstream event when the database is closed; it's not a mandatory. It requires a larger changes to the database to be able to send close events to the PassthroughSubject which I think we shouldn't do it in the patch release but in the minor or major release instead.